cdap-examples/WikipediaPipeline/src/main/scala/co/cask/cdap/examples/wikipedia/ScalaSparkLDA.scala

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.examples.wikipedia
+
+import co.cask.cdap.api.common.Bytes
+import co.cask.cdap.api.dataset.table.Table
+import co.cask.cdap.api.dataset.table.Put
+import co.cask.cdap.api.spark.{SparkContext, ScalaSparkProgram}
+import co.cask.cdap.api.workflow.Value
+import co.cask.cdap.api.workflow.WorkflowToken
+import org.apache.spark.{AccumulatorParam, Accumulator}
+import org.apache.spark.mllib.clustering.{LDAModel, LDA}
+import org.apache.spark.mllib.linalg.{Vector, Vectors}
+import org.apache.spark.rdd.{RDD, NewHadoopRDD}
+
+import java.text.BreakIterator
+import java.util
+
+import scala.collection.mutable
+
+/**
+ * Scala program to run Latent Dirichlet Allocation (LDA) on wikipedia data.
+ * This is an adaptation of
+ * https://github.com/apache/spark/blob/master/examples/src/main/scala/org/apache/spark/examples/mllib/LDAExample.scala
+ * for CDAP.
+ */
+class ScalaSparkLDA extends ScalaSparkProgram {
```

Comment:
Sure. I'm not too concerned though, since this is an (mostly, other than a few CDAP calls) adaptation of an example from spark source. If anyone wants to review, its more than welcome :)